nixos/modules/services/web-servers/nginx/location-options.nix

```
@@ -111,6 +112,21 @@ with lib;
       '';
     };
 
+    locations = mkOption {
+      type = types.attrsOf (types.hideSubTree (types.submodule self));
```

The `hideSubTree` hack is not necessary anymore thanks to @roberth with https://github.com/NixOS/nixpkgs/pull/143207/commits/64dfd983df61d9c427ac01d0d25b8d2f26b4c3d5. Now you can do

```suggestion
      type = types.attrsOf (types.submodule self);
      visible = "shallow";
```